src/durableorchestrationclient.ts

```diff
@@ -226,6 +226,9 @@ export class DurableOrchestrationClient {
             case 400: // instance failed or terminated
             case 404: // instance not found or pending
             case 500: // instance failed with unhandled exception
+                if (!response.data || !Object.keys(response.data).length) {
+                    return undefined;
+                }
                 return response.data as DurableOrchestrationStatus;
             default:
                 return Promise.reject(this.createGenericError(response));

```

Comment:
Is there a reason why you would prefer to get `undefined` instead of throwing an exception? The latter seems, to me, more helpful as it forces the end user to respond / deal with the exception, whereas `undefined` variables can cause silent errors.

Reply:
The idea was to handle this issue more gracefully, which would not break existing implementations. But I agree that throwing an exception, in this case, would be a more idiomatic approach.

Reply:
Even though a package update does not happen automatically, a user of this package expects a package maintainer to follow semver rules. This means introducing a breaking change by throwing an additional exception should be done in a major release.

In my opinion, this is a nasty bug that should be fixed sooner rather than later, e.g. in a patch release.

Reply:
People might be handling the empty string right now, so changing to undefined would break them as well. I agree this fix should be done in a major release, and I think all the solutions discussed so far are equally breaking.

The good news is we are actually about to announce public preview of the next major version. @hossam-nasr @davidmrdavid can we fix this for v3 ASAP?

Reply:
You are right people might have dealt with this issue in different ways as such either way it would break stuff for some or another.

Changed the fix to throw an error instead: [a95be58](https://github.com/Azure/azure-functions-durable-js/pull/487/commits/a95be58d4cbf5cfb612a58b6f8860c34fe727a49)

Reply:
Thanks for the fix @christianemmert.
Based on @ejizba's suggestion, I think this is a good candidate for the v3 version that's, as of today, in public preview. Let me re-target this PR for that branch.

Reply:
Hmm, seems that might not easy to do through the GitHub UI, due to the divergence between the two v2 and v3 branches. causing merge conflicts, so for now I'm keeping this branch as targeting v2. 

We might need to do some manual porting. @christianemmert - if you would like to continue driving this change, you're welcome to re-target this diff to V3, although it will require cherry-picking your commits. Otherwise, I'm happy to take over this PR and do that myself. Please let me know if you have a preference.